lib/src/validator/name.dart

```diff
@@ -10,21 +10,16 @@ import '../entrypoint.dart';
 import '../utils.dart';
 import '../validator.dart';
 
-/// A validator that validates the name of the package and its libraries.
+/// A validator that the name of a package is legal and matches the library name
+/// in the case of a single library.
 class NameValidator extends Validator {
   NameValidator(Entrypoint entrypoint) : super(entrypoint);
 
   Future validate() {
     return Future.sync(() {
-      _checkName(entrypoint.root.name, 'Package name "${entrypoint.root.name}"',
-          isPackage: true);
+      _checkName(entrypoint.root.name);
 
       var libraries = _libraries;
-      for (var library in libraries) {
-        var libName = path.basenameWithoutExtension(library);
-        _checkName(libName, 'The name of "$library", "$libName",',
-            isPackage: false);
-      }
```

Comment:
I think this check is valuable, but as you point out we're not running the analyzer here and we can't spot whether the file is using `part of`, so this check is too simple.

Hence, I agree we should omit it...